### PR TITLE
Check `mbstring.func_overload` only if the mb module is installed.

### LINKED ISSUE #14670: Check for mbstring.func_overload in lib/private/util.php fails if php-mbstring extension is not installed
Just ran into this on the latest fedora, was very hard to debug. We need to check if the mbstring module exists before checking for the func_overload setting

@LukasReschke @DeepDiver1975 @MorrisJobke @PVince81

### PR DESCRIPTION
**Pull request by LukasReschke**
Fixes #14670

cc @Raydiation Please test and review.
cc @karlitschek

### CODE REVIEW

File: lib/private/util.php [hunk 1]
```diff
@@ -740,13 +739,20 @@ public static function checkServer(\OCP\IConfig $config) {
 			$webServerRestart = true;
 		}
 
-		if (version_compare(phpversion(), '5.4.0', '<')) {
+		/**
+		 * The mbstring.func_overload check can only be performed if the mbstring
+		 * module is installed as it will return null if the checking setting is
+		 * not available and thus a check on the boolean value fails.
+		 *
+		 * TODO: Should probably be implemented in the above generic dependency
```
Comment (LukasReschke) [inline lib/private/util.php]: Somewhat hacky I admit it. Alternate proposals welcome.

**Comment by Raydiation** [comment]
Great, will test later on a new fedora image

**Comment by nickvergessen** [comment]
:+1: works

**Comment by karlitschek** [comment]
:+1:

**Comment by nickvergessen** [comment]
needs rebase

**Comment by LukasReschke** [comment]
Rebased.

**Pull request merged**